### PR TITLE
Add bonus script: kindle-display

### PR DESCRIPTION
**Pull request by dennisreimann**
Added my [Bitcoin Status Display](https://github.com/dennisreimann/kindle-display). Let me know if this needs more documentation or integration.

### CODE REVIEW

**Comment by Emzy** [comment]
Tested ACK

But without a btcpayserver there is no price displayed.

It uses a Firefox to render out (via cron every 5 minutes) the png file for the Kindle. Seems a bit heavy for the task.
But works on my 2GB Ram Raspberry Pi.

**Comment by openoms** [comment]
This is great, but setting up the Kindle is far from trivial (just taking a break from trying to make `usbNetwork` work ). Will make some notes if succeed at some point.

I'd like to suggest to put it in the `SETTINGS` menu with the other display related things.

Testing:
```
./XXsyncScripts.sh kindle-display dennisreimann
./config.scripts/bonus.kindle-display.sh on
```

**Comment by Emzy** [comment]
> This is great, but setting up the Kindle is far from trivial (just taking a break from trying to make `usbNetwork` work ). Will make some notes if succeed at some point.

Yes this was super hard, you need to find the right thing for the model and firmware version. It took me like 2 hours. Already forgot how I did it (was 4 weeks ago).

> I'd like to suggest to put it in the `SETTINGS` menu with the other display related things, rather than listing in `SERVICES` with the 1 click installs.

+1

**Comment by openoms** [comment]
Succes on Kindle 4 (v 4.1.4)
Rudimentary notes: https://gist.github.com/openoms/56979d0859d7063cb734bdacabf1068f

Files used:
image

File: home.admin/config.scripts/bonus.kindle-display.sh [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+if [ "$1" = "1" ] || [ "$1" = "on" ]; then
+  echo "*** INSTALL KINDLE-DISPLAY ***"
+
+  isInstalled=$(sudo ls $HOME_DIR 2>/dev/null | grep -c 'kindle-display')
+  if [ ${isInstalled} -eq 0 ]; then
+    # install dependencies
+    sudo apt install -y firefox-esr pngcrush jo
+
+    # install nodeJS
+    /home/admin/config.scripts/bonus.nodejs.sh
```
Comment (openoms) [inline home.admin/config.scripts/bonus.kindle-display.sh]: needs to be:
`/home/admin/config.scripts/bonus.nodejs.sh on`

Comment (dennisreimann) [reply home.admin/config.scripts/bonus.kindle-display.sh]: Thanks for the pointer, fixed!

**Comment by dennisreimann** [comment]
@openoms Ok, thanks for your suggestions. As discussed I'll leave this out of the menus as it is hard to seet up and I don't explicitely want to advertise usage of this. The main purpose of adding it was that some people were asking for it and it will make their life easier :)

Also [added your notes to the main README](https://github.com/dennisreimann/kindle-display#notes-regarding-usb-network-via-openoms) of my repo, others might find them helpful when setting up USB networking.

I consider this finished for now – it is also using a fixed version of the app instead of pulling the master branch on install.

**Comment by openoms** [comment]
After getting this work initially now I only have this image generated on my RPi:
image

Node version is: Node.js v12.16.3.

```
$ sudo systemctl status kindle-display.service 
● kindle-display.service - kindle-display
   Loaded: loaded (/etc/systemd/system/kindle-display.service; enabled; vendor preset: enabled)
   Active: active (running) since Sun 2020-07-05 08:36:16 BST; 3s ago
 Main PID: 15910 (npm)
    Tasks: 19 (limit: 4915)
   Memory: 19.7M
   CGroup: /system.slice/kindle-display.service
           ├─15910 npm
           ├─15925 sh -c node -r dotenv/config index.js
           └─15926 node -r dotenv/config index.js

Jul 05 08:36:16 raspberrypi systemd[1]: Started kindle-display.
Jul 05 08:36:17 raspberrypi npm[15910]: > kindle-display@0.2.0 start /home/kindledisplay/kindle-display/server
Jul 05 08:36:17 raspberrypi npm[15910]: > node -r dotenv/config index.js
Jul 05 08:36:18 raspberrypi npm[15910]: Running at http://localhost:3030
```

Tried reinstalling, restarting, but not sure what is wrong.

**Comment by dennisreimann** [comment]
@openoms this looks like the [initial data](https://github.com/rootzoll/raspiblitz/pull/1301/files#diff-9ace2ed4f37dbe8ede11a527fcbc90e6R77) could not get generated. Not sure why this might have failed. I tried running off/on and for me it worked, but I'll submit a fix PR that ensures the file is there so that the server can start.

**Comment by dennisreimann** [comment]
Addressed in #1316

**Comment by grnqrtr** [comment]
Just thought I would comment that this is working well for me too. I had some trouble with the USBNetwork setup, even with @openoms extra instructions that you added. This is what I eventually found and was successful with (https://www.mobileread.com/forums/showthread.php?t=204942):

```

-     un-plug cable (if still plugged in)
-     toggle USBnetwork ON in launcher
-     plug the cable
-     kill any automation (or configure yours to do: )
-     sudo ip link set up dev usb0 (It may already be up)
-     sudo ip address add 192.168.15.201 peer 192.168.15.244 dev usb0
-     use the networking until your done (telnet 192.168.15.244)
-     un-plug cable
-     toggle USBnetwork OFF in launcher

```

**Comment by dennisreimann** [comment]
@grnqrtr Thanks, I've added it to the README.

Also a heads-up for everyone: With #1321 and the kindle-display v0.4.0 I've refined this a bit more, which allows to choose the exchange rates and require outside connections to be done via Tor. Thanks @openoms and @csperle for the contributions that lead to this! 🍰

**Pull request merged**